handlers/handler-fault-injection/src/main/java/org/apache/servicecomb/faultinjection/FaultInjectionHandler.java

```diff
@@ -0,0 +1,222 @@
+          "abort.percent",
+          FAULT_INJECTION_ABORT_PERCENTAGE_DEFAULT);
+
+      // check fault delay condition.
+      boolean isAbort = checkFaultInjectionDelayAndAbort(reqCountCurrent, abortPercent);
+      if (isAbort) {
+        // get the config values related to delay percentage.
+        int errorCode = getFaultInjectionConfig(invocation,
+            "abort.httpStatus",
+            FAULT_INJECTION_ABORT_ERROR_MSG_DEFAULT);
+        // if request need to be abort then return failure with given error code
+        CommonExceptionData errorData = new CommonExceptionData("aborted by fault inject");
+        asyncResp.consumerFail(
+            new InvocationException(new HttpStatus(errorCode, "aborted by fault inject"), errorData));
+        return;
+      }
+    }
+
+    // if no delay and no abort then continue to next handler.
+    invocation.next(asyncResp);
+  }
+
+  /**
+   * It will check the delay/abort condition based on request count and percentage
+   * received.
+   * 
+   * @param reqCount
+   * @param percentage
+   * @param key
+   * @return true/false
+   */
+  private boolean checkFaultInjectionDelayAndAbort(long reqCount, int percentage) {
+    /*
+     * Example: delay/abort percentage configured is 10% and Get the count(suppose
+     * if it is 10th request) from map and calculate resultNew(10th request) and
+     * requestOld(9th request). Like this for every request it will calculate
+     * current request count and previous count. if both not matched need to add
+     * delay/abort otherwise no need to add.
+     */
+
+    // calculate the value with current request count.
+    long resultNew = (reqCount * percentage) / 100;
+
+    // calculate the value with previous count value.
+    long resultOld = ((reqCount - 1) * percentage) / 100;
+
+    // if both are not matching then delay/abort should be added.
+    if (resultNew != resultOld) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Handles the reading fault injection configuration.
+   * 
+   * @param invocation
+   *            invocation of request
+   * @param key
+   *            configuration key
+   * @return configuration value
+   */
+  private int getFaultInjectionConfig(Invocation invocation, String key, int defaultValue) {
+    int value = 0;
+    String config;
+    FaultInjectionConfig faultCfg = new FaultInjectionConfig();
+
+    //first need to check in config center map which has high priority.
+    Map<String, AtomicInteger> cfgMap = FaultInjectionUtil.getConfigCenterMap();
+
+    // get the config base on priority. operationName-->schema-->service-->global
+    String operationName = invocation.getOperationName();
+    String schema = invocation.getSchemaId();
+    String serviceName = invocation.getMicroserviceName();
+
+    if (operationName != null && schema != null && serviceName != null) {
+      if (invocation.getTransport().getName().equals(FAULTINJECTION_HIGHWAY_TRANSPORT)) {
```

Comment:
why not just use invocation.getTransport().getName()

if use the "if" logic, and if add new transport in the furture,  this place should be modified too.